meta-hailo/README.md

```diff
@@ -0,0 +1,25 @@
+# meta-hailo - Support for HAILO NPU chips
+
+This repo provides recipes to build the kernel and userspace drivers and utilities
+for the Hailo-8 NPU chip.
+This includes the following main components:
+
+- hailo-pci (OSS kernel module)
+- hailo-firmware (proprietary firmware for hailo8 chip)
+- hailort (Userspace API)
+  - hailortcli (tool)
+  - hailort (service)
+  - libhailort (c library)
+  - libhailort-dev (dev package)
+  - python3-hailort (python library)
+
+> Note: `numpy==1.23.3` is preinstalled. Please do not upgrade it, as
+> `hailort v4.18.0` requires this specific version. Upgrading `numpy` may cause
+> `pyhailort` to fail to execute.
```

Comment:
Will apt-upgrade updates it? If do, I bet you need a package pin config.

Reply:
You would need to pin it via apt preferences then. But that is security-wise a bad idea as well, thus this issue of hailo stack needs fixing.

Reply:
Currently, we are using Hailo V4.18, which officially requires numpy 1.23.3. Even if Hailo fixes this issue, they may not backport the fix to V4.18, making the situation quite challenging. There are significant differences between numpy 1.23 and 1.24, meaning that backporting the patches to V4.18 would require a substantial effort.

So far, IMHO, it would be a good choice for us.

Reply:
We are already patching hailo for other reasons. Key question is what is breaking when using a (distro-)maintained numpy version in their stack?

Reply:
I believe the issue is caused by the changes mentioned in the numpy 1.24.0 release notes:
 https://numpy.org/doc/2.0/release/1.24.0-notes.html
```md
There are also a large number of new and expired deprecations due to changes in promotion and cleanups. This might be called a deprecation release. Highlights are

    Many new deprecations, check them out.

    Many expired deprecations,

    New F2PY features and fixes.

    New “dtype” and “casting” keywords for stacking functions.
```

Reply:
I suppose Hailort will require a bunch of modifications to eliminate the deprecations.